reactioncommerce/reaction
# Fix: Changing the permalink before publishing a product results in "not found"

## Issue
Closes #3858

If the permalink of a product is changed, but not published yet, the permalink is assembled with the new (yet unpublished) handle, which leads to the PDP rendering a "Not found" template.
See this animated gif:
![unpulished_notfound](https://user-images.githubusercontent.com/1733229/36217896-e54e873c-11b2-11e8-9d1b-bdcbe66dcf3e.gif)


## What it does:
- Prevent unpublished handle changes to be used when assembling a link
URL to PDP.

## What is does not:
- Ensure that the "not found" template is rendered in certain situations
(which I don't know what these are exactly) AFTER publishing a product. This is a separate issue: #2249

## Solution
If a product has unpublished changes, they published state is in
products.__published. Use that for building the perma-link from
products.__published.handle.

Impact: major  
Type: bugfix

## Breaking changes
not known


## Testing
1. As a admin, create a new product
2. Change the handle (permalink)
3. Do not publish
4. navigate to product grid
5. navigate into PDP again
6. The product detail page should be shown correctly
